### ISSUE TITLE
Torrent fetch on Synology results in Python exception

### ISSUE THREAD

**Issue by stephanmeesters** [SickGear/SickGear#1172]
Trying to fetch a torrent results in an error printed in the log. I'm running a Synology NAS with DSM 6.2.1 and am using the SickGear master branch.

I'm not sure whether it is somehow a permissions issue with my NAS. I've tried disabling/enabling all kinds of security features on DSM to try to make this work with no avail.

The 'sc-download' user group has access to my videos shared folder and has permissions to use the SearBeard-custom app, etc. The Python and Git modules were installed from SynoCommunity according to the instructions.

The "Search torrents" is set to enable and the connection to Synology DS checks out.

Here is the log file including the Python exception:

```
2019-04-11 22:55:18 INFO     BACKLOG-328487 :: Completed backlog search for: [The Orville]
2019-04-11 22:55:18 INFO     BACKLOG-328487 :: No needed episodes found during backlog search for: [The Orville]
2019-04-11 22:55:18 INFO     BACKLOG-328487 :: No suitable candidates
2019-04-11 22:55:18 INFO     BACKLOG-328487 :: [The Pirate Bay] :: No episode items in response from https://thepiratebay-org6-prox26-info.unblocked.is/search/The.Orville%20S02E01%2A%7C2x01%2A/0/7/200
2019-04-11 22:55:18 INFO     BACKLOG-328487 :: [The Pirate Bay] :: Performing episode search for The Orville
2019-04-11 22:55:18 WARNING  BACKLOG-328487 :: Exception caught while loading URL http://itorrents.org/torrent/A0275B190F90B14C4CE50C1A61423A12FE10324B.torrent?title=The-Orville-S02E01-1080p-WEB-x264-TBS
 1) Detail... 'NoneType' object has no attribute 'ok'
 2) Traceback (most recent call last):
 3)   File "/volume1/@appstore/sickbeard-custom/var/SickBeard/sickbeard/helpers.py", line 1218, in getURL
 4)     if response.ok and not response.content and 'url=' in response.headers.get('Refresh', '').lower():
 5) AttributeError: 'NoneType' object has no attribute 'ok'

2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is HD TV for [The.Orville.S02E01.720p.x264-StB]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.Jaloja.720p.AMZN.WEB-DL.DDP5.1.H.264-NTb[TGx]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.720p.WEB.x264-TBS[TGx]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.720p.WEB.x264-300MB]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.720p.WEB.x264-TBS[ettv]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.Jaloja.720p.AMZN.WEBRip.DDP5.1.x264-NTb[rarbg]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.Jaloja.720p.AMZN.WEB-DL.DDP5.1.H.264-NTb[eztv]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [the.orville.s02e01.720p.web.x264-tbs[eztv]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 720p WEB-DL for [The.Orville.S02E01.720p.WEB.x264-TBS]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 1080p WEB-DL for [The.Orville.S02E01.1080p.WEB.x264-TBS[TGx]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 1080p WEB-DL for [The.Orville.S02E01.Jaloja.1080p.AMZN.WEB-DL.DDP5.1.H.264-NTb[TGx]]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 1080p WEB-DL for [The.Orville.S02E01.1080P.WEB-DL.DD5.1.H.264]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 1080p WEB-DL for [The.Orville.S02E01.Jaloja.1080p.AMZN.WEBRip.DDP5.1.x264-NTb]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: Quality is 1080p WEB-DL for [The.Orville.S02E01.1080p.WEB.x264-TBS]
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: [LimeTorrents] :: 20 episode candidates
2019-04-11 22:55:17 INFO     BACKLOG-328487 :: [LimeTorrents] :: 35 episode items in response from https://www.limetorrents.cc/search/tv/The.Orville+S02E01/
2019-04-11 22:55:12 INFO     BACKLOG-328487 :: [LimeTorrents] :: Performing episode search for The Orville
2019-04-11 22:55:12 INFO     BACKLOG-328487 :: Beginning backlog search for: [The Orville]
2019-04-11 22:55:09 INFO     TORNADO :: Starting backlog search for The Orville season 2 because eps were set to wanted
2019-04-11 22:33:18 INFO     RECENT-SEARCH :: No search of cache for episodes required
2019-04-11 22:33:18 WARNING  RECENT-SEARCH :: No NZB/Torrent providers in Media Providers/Options are enabled to match recent episodes
2019-04-11 22:33:18 INFO     RECENT-SEARCH :: No unaired episodes marked wanted
2019-04-11 22:28:30 INFO     FETCH-XEMDATA :: Finished fetching show ids with 954 xem scene mappings for origin TheTVDB
2019-04-11 22:28:29 INFO     UPGRADE-HISTORY-ACTION :: Updating history items with status Snatched Proper in a background process...
2019-04-11 22:28:28 INFO     FETCH-XEMDATA :: Fetching show ids with xem scene mappings for origin TheTVDB
2019-04-11 22:28:16 INFO     CHECKVERSION :: No update needed
2019-04-11 22:28:15 INFO     MAIN :: Updating timezone info with new one: zoneinfo-2019a.tar.gz
2019-04-11 22:28:14 INFO     CHECKVERSION :: Checking if git needs an update
2019-04-11 22:28:14 INFO     LOAD-MAPPINGS :: Indexer mappings loaded
2019-04-11 22:28:14 INFO     LOAD-MAPPINGS :: Start loading Indexer mappings...
2019-04-11 22:28:14 INFO     TORNADO :: Starting SickGear on http://0.0.0.0:8083/
2019-04-11 22:28:14 INFO     MAIN :: Loading initial show list
2019-04-11 22:28:14 INFO     MAIN :: Writing PID: 3331 to /volume1/@appstore/sickbeard-custom/var/sickbeard-custom.pid
2019-04-11 22:28:14 INFO     MAIN :: No UNAIRED episodes, check passed
2019-04-11 22:28:14 INFO     MAIN :: No orphan episodes, check passed
2019-04-11 22:28:14 INFO     MAIN :: No duplicate episode, check passed
2019-04-11 22:28:14 INFO     MAIN :: No duplicate show, check passed
2019-04-11 22:28:14 INFO     MAIN :: Updating TV Episode table with index idx_tv_ep_ids
2019-04-11 22:28:14 INFO     MAIN :: Database upgrade required: History Status
2019-04-11 22:28:14 INFO     MAIN :: Database upgrade required: History
2019-04-11 22:28:14 INFO     MAIN :: Database upgrade required: Size And Provider
2019-04-11 22:28:14 INFO     MAIN :: Database upgrade required: Initial Schema
2019-04-11 22:28:14 INFO     MAIN :: Checking database structure...
2019-04-11 22:28:13 INFO     MAIN :: Database upgrade required: Initial Schema
2019-04-11 22:28:13 INFO     MAIN :: Checking database structure...
2019-04-11 22:28:13 INFO     MAIN :: Proceeding with upgrade
2019-04-11 22:28:13 INFO     MAIN :: Backing up database before upgrade
```

The error log of SickGear shows the following error, which I'm not sure is related to the issue:

```
Please make sure you have the correct access rights
and the repository exists.
2019-04-11 22:38:54 TORNADO :: git ls-remote --heads None returned : fatal: 'None' does not appear to be a git repository
fatal: Could not read from remote repository.
```

**Comment by JackDandy** [comment]
The steps section requested when this issue was created, enables your reported issue to be replicated and investigated, however, you removed it so there is no idea how to trigger your reported event in order to fix it for you. Until you can supply this information, then there is no choice but to mark this issue "invalid" as what you are reporting cannot be verified. Sorry.

Essential help on [[How-to] Report Issues](https://github.com/SickGear/SickGear/wiki/%5BHow-to%5D-Report-Issues)

There's no need to open a new issue, additions to the end of this thread are just fine.

Thanks.

**Comment by JackDandy** [comment]
If you install Transmission/qBittorent, and instead of using blackhole connect either directly, this issue will no longer occur again.
